### PR TITLE
Ft develop join ride #158498887

### PR DESCRIPTION
**Pull request by Quantum-35**
**What does this PR do?**
Finishes the endpoint  where **Registered** and and **Logged** in user can *POST* a join request to available rides
Uses JWT tokens to restict access to this endpoint by unregistered users or users who have not logged in
The Live version of this application can be found [here](https://challenge-ride-my-way.herokuapp.com)
**To test it**
```
1.Copy the link to Postman
2.Signup by appending */api/v1/auth/register* to the heroku link
3.Login by appending */api/v1/auth/login* to the heroku link
4.Make ride offers by sending Post to */api/v1/rides/* 
5. Make a join request by sending Post to */api/v1/1/request*
```

**Description of Task to be completed**
Implement Pep8 and increase test coverage

##If you wish to clone it
Clone this repository to a local folder
Navigate to the cloned repo and Run:
On Linux or macos terminal

    python3 -m venv venv
    . venv/bin/activate

Or on Windows cmd::

    py -3 -m venv venv
    venv\Scripts\activate.bat

Install the dependencies::

    pip install -r requirements.txt 
    *Then git  checkkout ft-develop join ride #158498887
Run the development server:
  ```
  $ python run.py
  ```
**To run test with coverage**
pytest --cov-report term-missing --cov=app
**What PT stories?**
https://www.pivotaltracker.com/n/projects/2179783

### CODE REVIEW

File: api/v1/tests/test_helper.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+from app.auth.helper import (email_validator,
+                             address_validator,
+                             password_validator,
+                             user_name_validator)
+
+
+class TestHelper(unittest.TestCase):
+
+    def test_valid_email(self):
+        self.assertTrue(email_validator('quantum@gmail.com'))
```
Comment (JoshuaOndieki) [inline api/v1/tests/test_helper.py]: Suggested missing edge case:
- Test whether an invalid email is valid according to your validator.

Comment (JoshuaOndieki) [reply api/v1/tests/test_helper.py]: Do so for the other validators too.

Comment (Quantum-35) [reply api/v1/tests/test_helper.py]: Thanks for your feedback.I am implementing it
